solutions/src/main/java/SimpleFxConversionService.java

```diff
@@ -0,0 +1,66 @@
+import ru.sberbank.school.task02.ExternalQuotesService;
+import ru.sberbank.school.task02.FxConversionService;
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Quote;
+import ru.sberbank.school.task02.util.Symbol;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import static ru.sberbank.school.task02.util.ClientOperation.BUY;
+
+public class SimpleFxConversionService implements FxConversionService {
+
+    private ExternalQuotesService externalQuotesService;
+
+    public SimpleFxConversionService(ExternalQuotesService externalQuotesService) {
+        this.externalQuotesService = externalQuotesService;
+    }
+
+    @Override
+    public BigDecimal convert(ClientOperation operation, Symbol symbol, BigDecimal amount) {
+        if (operation == null) {
+            throw new IllegalArgumentException("operation is not defined");
+        }
+
+        if (symbol == null) {
+            throw new IllegalArgumentException("instrument is not defined");
+        }
+
+        if (amount == null || (amount.compareTo(BigDecimal.ZERO) <= 0)) {
+            throw new IllegalArgumentException("amount is not defined or not positive");
+        }
+
+        List<Quote> quotes = externalQuotesService.getQuotes(symbol);
+        Quote quote = null;
+        for (Quote quoteCandidate : quotes) {
+            if (quoteCandidate == null) {
+                continue;
+            }
+
+            if (quote == null) {
+                if (quoteCandidate.isInfinity()) {
+                    quote = quoteCandidate;
+                }
```

Comment:
&#39;}&#39; at column 17 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-193772611>*